### PR TITLE
Issue 83 - Hide social like for unpublished content

### LINKED ISSUE #83: Hide social like for unpublished content
One of our customer complain that facebook has a high share number for unpublished content:

highshare

It looks like facebook has a bug that it treat every unpublished post as the same post, and counts on.

To hide false positives we should hide social like for unpublished content.

### PR DESCRIPTION
**Pull request by rodfersou**
closes #83 
Needs to:
- [x] Create tests
- [x] Test implementation with Plone 5

### CODE REVIEW

**Comment by hvelarde** [comment]
why not show the viewlet only if the content is published?

**Comment by rodfersou** [comment]
@hvelarde read the issue please

**Comment by rodfersou** [comment]
Review code on Plone 5 generated html

File: setup.py [hunk 1]
```diff
@@ -64,6 +64,7 @@
       ],
       extras_require={
           'test': [
+              'mock',
```
Comment (rodfersou) [inline setup.py]: needed for python 2.7.12

**Comment by hvelarde** [comment]
it is better to keep the viewlet hidden until the content has been published.

**Comment by hvelarde** [comment]
please, double check as locally I can see the following issue:

```python
Error in test test_disabled_on_portal (sc.social.like.tests.test_viewlets.LikeViewletTestCase)
Traceback (most recent call last):
  File "/home/hvelarde/collective/buildout.python/parts/opt/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/hvelarde/collective/sc.social.like/sc/social/like/tests/test_viewlets.py", line 83, in test_disabled_on_portal
    self.assertFalse(viewlet.enabled())
  File "/home/hvelarde/collective/sc.social.like/sc/social/like/browser/viewlets.py", line 45, in enabled
    api.content.get_state(self.context) == 'published',
  File "<string>", line 2, in get_state
  File "/home/hvelarde/.buildout/eggs/plone.api-1.5.1-py2.7.egg/plone/api/validation.py", line 70, in wrapped
    return f(*args, **kwargs)
  File "/home/hvelarde/.buildout/eggs/plone.api-1.5.1-py2.7.egg/plone/api/content.py", line 358, in get_state
    return workflow.getInfoFor(ob=obj, name='review_state')
  File "/home/hvelarde/.buildout/eggs/Products.CMFCore-2.2.9-py2.7.egg/Products/CMFCore/WorkflowTool.py", line 263, in getInfoFor
    raise WorkflowException(msg)
WorkflowException: No workflow provides '${name}' information.
```

**Comment by hvelarde** [comment]
it's also happening on Travis: https://travis-ci.org/collective/sc.social.like/jobs/185199926#L1473

File: sc/social/like/testing.py [hunk 2]
```diff
@@ -46,6 +46,7 @@ def setUpPloneSite(self, portal):
             self.applyProfile(portal, 'collective.cover:default')
 
         self.applyProfile(portal, 'sc.social.like:default')
+        portal.portal_workflow.setDefaultChain('simple_publication_workflow')
```
Comment (rodfersou) [inline sc/social/like/testing.py]: isn't it supposed to break other tests where content are not published (like whatsapp and telegram robot tests)?

File: sc/social/like/testing.py [hunk 3]
```diff
@@ -46,9 +46,7 @@ def setUpPloneSite(self, portal):
             self.applyProfile(portal, 'collective.cover:default')
 
         self.applyProfile(portal, 'sc.social.like:default')
-        portal_workflow = portal.portal_workflow
-        portal_workflow.setChainForPortalTypes(
-            ['Document'], ['simple_publication_workflow'])
+        portal.portal_workflow.setDefaultChain('simple_publication_workflow')
```
Comment (rodfersou) [inline sc/social/like/testing.py]: isn't it supposed to break other tests where content are not published (like whatsapp and telegram robot tests)?

Comment (hvelarde) [inline sc/social/like/testing.py]: those tests are not working in the first place.

Comment (rodfersou) [inline sc/social/like/testing.py]: fine

**Pull request merged**